collector.go

```diff
@@ -348,17 +348,17 @@ func collectGenericSensor(ch chan<- prometheus.Metric, state float64, data senso
 	)
 }
 
-func (c collector) collectMonitoring(ch chan<- prometheus.Metric, creds Credentials) error {
+func (c collector) collectMonitoring(ch chan<- prometheus.Metric, creds Credentials) int {
 	output, err := ipmiMonitoringOutput(c.target, creds.User, creds.Password)
 	if err != nil {
-		log.Errorln(err)
-		return err
+		log.Errorf("Failed to collect ipmimonitoring data: %s", err)
+		return 0
```

Comment:
I would probably just remove the `return` here for now, this avoids dropping data for later collectors if an earlier collector fails.

Reply:
This I don't understand. This function is only a single collector, `return`ing from this only aborts this specific collector. The other collectors don't use any data from this one. Can you elaborate?